lib/spack/spack/test/packaging.py

```diff
@@ -334,6 +349,25 @@ def test_elf_paths():
     assert out == ['/opt/lib', '/opt/lib64', '/opt/local/lib']
 
 
+@pytest.mark.skipif(sys.platform == 'darwin',
+                    reason="only works with ELF objects")
+def test_empty_rpath(tmpdir):
+    patchelf = has_patchelf()
+    if patchelf:
+        paths = get_existing_elf_rpaths('/bin/bash')
+        assert (paths is None)
+        relocate_binary(('/bin/bash', 'bin/sh'), '/lib', '/opt/lib')
+        paths = get_existing_elf_rpaths('/bin/bash')
+        assert (paths is None)
+        with tmpdir.as_cwd():
+            shutil.copyfile('/bin/perl', 'perl')
```

Comment:
I'm worried this will fail if `perl` isn't available. It may be more robust to create an RPATH'ed binary manually. That being said I don't think that has to be sorted out here - as long as we are testing `get_existing_elf_rpaths` and `get_relative_rpaths` then we are covering most of the logic up to the invocation of the `patchelf` command to actually update the binary.

As a side note though: the method signature for `modify_elf_object` is

```
modify_elf_object(path_name, orig_rpath, new_rpath)
```

but the `orig_rpath` parameter is not actually used in the method.